pycytominer/cyto_utils/cell_locations.py

```diff
@@ -0,0 +1,401 @@
+
+        else:
+            return obj
+
+    def _parse_s3_path(self, s3_path):
+        """Parse an S3 path into a bucket and key
+
+        Parameters
+        ----------
+        s3_path : str
+            The S3 path
+
+        Returns
+        -------
+        str
+            The bucket
+        str
+            The key
+        """
+
+        s3_path = s3_path.replace("s3://", "")
+
+        bucket = s3_path.split("/")[0]
+
+        key = "/".join(s3_path.split("/")[1:])
+
+        return bucket, key
+
+    def _s3_file_exists(self, s3_path):
+        """Check if a file exists on S3
+
+        Parameters
+        ----------
+        s3_path : str
+            The path to the file on S3
+
+        Returns
+        -------
+        bool
+            True if the file exists on S3, False otherwise
+        """
+
+        s3 = boto3.resource("s3")
+
+        bucket, key = self._parse_s3_path(s3_path)
+
+        try:
+            s3.Object(bucket, key).load()
+        except botocore.exceptions.ClientError as e:
+            if e.response["Error"]["Code"] == "404":
+                return False
+            else:
+                raise
+        else:
+            return True
+
+    def _download_s3(self, uri):
+        """
+        Download a file from S3, save it to a temporary directory, and return the path to the file
+        """
+        s3 = boto3.resource("s3")
+
+        bucket, key = self._parse_s3_path(uri)
+
+        tmp_dir = tempfile.mkdtemp()
+        tmp_file = pathlib.Path(tmp_dir) / pathlib.Path(key).name
+
+        s3.Bucket(bucket).download_file(key, tmp_file)
+
+        return tmp_file
+
+    def _load_metadata(self):
+        """Load the metadata into a Pandas DataFrame
+
+        Returns
+        -------
+        Pandas DataFrame
+            The metadata loaded into a Pandas DataFrame
+        """
+
+        if not isinstance(self.metadata_input, pd.DataFrame):
+            # verify that the metadata file is a CSV or a Parquet file
+
+            if not (
+                self.metadata_input.endswith(".csv")
+                or self.metadata_input.endswith(".parquet")
+            ):
+                raise ValueError("Metadata file must be a CSV or a Parquet file")
+
+            # load the metadata file into a Pandas DataFrame
+            if self.metadata_input.endswith(".csv"):
+                df = pd.read_csv(self.metadata_input, dtype=str)
+            else:
+                df = pd.read_parquet(self.metadata_input)
+                # cast all columns to string
+                df = df.astype(str)
+        else:
+            df = self.metadata_input
+
+        # verify that the image index columns are present in the metadata object
+
+        if not all(elem in df.columns for elem in self.image_index):
+            raise ValueError(
+                f"Image index columns {self.image_index} are not present in the metadata file"
+            )
+
+        return df
+
+    def _convert_to_per_row_dict(self, df):
```

Comment:
Would it make sense to name this method in a way that indicates the conversion to a dictionary-orientated dataframe (and not strictly a dictionary)?